File: tests/fixtures/python/clean.py
"""A clean file with no smells. Should produce zero findings."""


def add(a, b):
    return a + b


def greet(name):
    return f"Hello, {name}"


class Calculator:
    def __init__(self, value=0):
        self.value = value

    def add(self, n):
        self.value += n
        return self

    def subtract(self, n):
        self.value -= n
        return self

    def result(self):
        return self.value

File: tests/fixtures/cpp/clean.cpp
int add(int a, int b) {
    return a + b;
}

class Calculator {
public:
    Calculator(int value) : value_(value) {}

    int add(int n) {
        this->value_ += n;
        return this->value_;
    }

    int subtract(int n) {
        this->value_ -= n;
        return this->value_;
    }

    int result() {
        return this->value_;
    }

private:
    int value_;
};
